lib/browser.js

```diff
@@ -143,7 +143,7 @@ class Browser {
   }
 
   onDisconnect (disconnectedSocket) {
-    this.activeSockets.splice(this.activeSockets.indexOf(disconnectedSocket), 1)
```

Comment:
Hmm, it looked incorrect. When `disconnectedSocket` is not in `this.activeSockets` array then `this.activeSockets.indexOf(disconnectedSocket)` returns `-1`.

    array.splice(-1, 1)

remove last element of array.

I'll analyze unit tests

Reply:
`onDisconnect` method was called in unit test with two params instead of one, I fixed it

Reply:
The disconnect event sends a reason:
https://socket.io/docs/client-api/#Event-%E2%80%98disconnect%E2%80%99
But don't pass it through our handler.  Can we pass it in and append it to the this.disconnect() call? 

We see rare disconnect fails and better logging may help sort out why.